packages/admin/components/forms/ItemTypeahead/ItemTypeahead.tsx

```diff
@@ -1,77 +1,103 @@
-import React, { Ref, useMemo } from "react";
+import React, { Ref, useState } from "react";
 import { Controller } from "react-hook-form";
+import { QueryWrapper } from "@wdp/lib/api/components";
 import { graphql } from "react-relay";
 import type { FieldValues, Control, Path } from "react-hook-form";
 import { useTranslation } from "react-i18next";
-import Typeahead from "components/forms/Typeahead";
+import debounce from "lodash/debounce";
+import BaseTypeahead from "components/forms/BaseTypeahead";
+import {
+  ItemTypeaheadQuery as Query,
+  ItemTypeaheadQueryResponse as Response,
+} from "__generated__/ItemTypeaheadQuery.graphql";
 
-import type {
-  ItemTypeaheadFragment$data,
-  ItemTypeaheadFragment$key,
-} from "__generated__/ItemTypeaheadFragment.graphql";
-import { useMaybeFragment } from "hooks";
-type TypeaheadProps = React.ComponentProps<typeof Typeahead>;
+type TypeaheadProps = React.ComponentProps<typeof BaseTypeahead>;
 
 const ItemTypeahead = <T extends FieldValues = FieldValues>(
-  { data, control, name, label, disabled, required }: Props<T>,
+  { control, name, label, disabled, required }: Props<T>,
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   ref: Ref<HTMLInputElement>
 ) => {
-  const optionsData = useMaybeFragment(fragment, data);
+  const [q, setQ] = useState("a");
 
-  const options = useMemo(() => {
-    const options = optionsData?.viewer.items.nodes.map((node: ItemNode) => {
+  const formatOptions = (data: Response) => {
+    if (!data || !data.search?.results?.edges?.length) return;
+
+    const results = data.search.results.edges;
+    const options = results.map(({ node }) => {
       return {
         label: node.title || "",
-        value: node.id,
+        value: node.entity.id ?? "",
       };
     });
-
     return options;
-  }, [optionsData]);
+  };
 
   const { t } = useTranslation();
 
-  return options ? (
-    <Controller<T>
-      name={name}
-      control={control}
-      rules={{
-        validate: (value) => {
-          return !!value || (t("forms.validation.item") as string);
-        },
+  const debouncedOnChange = debounce((value) => setQ(value), 500);
+
+  return (
+    <QueryWrapper<Query> query={query} initialVariables={{ query: q }}>
```

Comment:
These changes work well, but I am concerned that the entire Controller and BaseTypeahead might be getting re-rendered with every query when just the list of options is changing. I would see if these components might benefit from using the `useQuery` hook with a memoized options list rather than the `QueryWrapper` component, if there's time.

Reply:
Hmm, I think you're right. My assumption was that moving the value to a hook and constant, rather than a wrapper component, would prevent Controller from re-rendering. I'm OK with leaving it as is and keeping an eye on performance issues.